runtime/src/iree/hal/drivers/cuda/cuda_device.c

```diff
@@ -234,6 +237,80 @@ static iree_status_t iree_hal_cuda_device_query_i64(
       (int)category.size, category.data, (int)key.size, key.data);
 }
 
+// Returns true if |id| is all zeros indicating an empty ID.
+static bool iree_hal_cuda_nccl_id_is_empty(const iree_hal_cuda_nccl_id_t* id) {
+  for (iree_host_size_t i = 0; i < IREE_ARRAYSIZE(id->data); ++i) {
+    if (id->data[i] != 0) return false;
+  }
+  return true;
+}
+
+static iree_status_t iree_hal_cuda_device_create_channel(
+    iree_hal_device_t* base_device, iree_hal_queue_affinity_t queue_affinity,
+    iree_hal_channel_params_t params, iree_hal_channel_t** out_channel) {
+  iree_hal_cuda_device_t* device = iree_hal_cuda_device_cast(base_device);
+
+  // TODO(#9580): check if nccl symbols are available - if not then we fail
+  // here and have the error propagated up to users. If we wanted to delay load
+  // NCCL we'd want to take a lock here, load it, and merge the symbols into the
+  // dynamic symbol table.
+  if (true) {
+    return iree_make_status(
+        IREE_STATUS_UNIMPLEMENTED,
+        "NCCL unavailable and collective operations cannot be performed");
```

Comment:
Ok cool, I was going to ask about this - there should probably be some minimum version check and channel creation should fail if using an older version / version without support, but checking for nccl symbols makes sense too.

Reply:
it'll take some time for collectives to work (compiler still needs work, runtime needs implementation) and it'll be unlikely that any extant compiler can target a (current) runtime with them - if this were a year from now we'd definitely want to gate it, though!